src/console_cmds.cpp

```diff
@@ -1049,6 +1049,54 @@ DEF_CONSOLE_CMD(ConNewGame)
 	return true;
 }
 
+DEF_CONSOLE_CMD(ConNewHeightMapGame)
+{
+	bool heightmap_loaded = true;
+
+	if (argc == 0) {
+		IConsoleHelp("Load a game by name or index. Usage: 'newheightmapgame [<file | number> [seed]]'");
+		IConsoleHelp("The server can force a new game using 'newheightmapgame'; any client joined will rejoin after the server is done generating the new game.");
+	}
+
+	if (argc >=1 && argc <= 3) {
+		const char *file = NULL;
+
+		if (argc == 1) {
+			file = _file_to_saveload.title;
+		}
+
+		if (argc >= 2) {
+			file = argv[1];
+		}
+
+		/* ConLoad command actually uses
+		_console_file_list.ValidateFileList()
+		here, but for this procedure build a file list for mere savegames, instead of merely validating */
+		_console_file_list.BuildFileList(FT_HEIGHTMAP, SLO_LOAD);
+		const FiosItem *item = _console_file_list.FindItem(file);
+		if (item != NULL) {
+			if (GetAbstractFileType(item->type) == FT_HEIGHTMAP) {
+				_switch_mode = SM_LOAD_HEIGHTMAP;
+				_file_to_saveload.SetMode(item->type);
+				_file_to_saveload.SetName(FiosBrowseTo(item));
+				_file_to_saveload.SetTitle(item->title);
+
+				StartNewHeightMapGameWithoutGUI((argc == 3) ? strtoul(argv[2], NULL, 10) : GENERATE_NEW_SEED);
+			} else {
+				IConsolePrintF(CC_ERROR, "%s: Not a heightmap.", file);
+			}
+		} else {
+			IConsolePrintF(CC_ERROR, "%s: No such file or directory.", file);
+		}
+
+		_console_file_list.InvalidateFileList();
+	}
+
+	if (argc > 3) heightmap_loaded = false;
```

Comment:
Why is this handled all the way down here?

Reply:
I was sorting blocks by increasing number of arguments, considering this case would be less frequent than normal ones, thus concluding it would avoid an usually unecessary extra check for those cases, making the program perform better on average.

Reply:
Seriously premature optimization there...

Reply:
Well I do not mind moving that up at all, judgments set aside. Where would that be, exactly?

Reply:
Move it before the `if (argc >=1 && argc <= 3) {` test, and make it return early. That means the rest of the function doesn't need to be in an if condition and can be unindented a bit.

Reply:
Done.